pkg/kv/kvserver/rangefeed/buffered_sender.go

```diff
@@ -10,95 +10,117 @@ import (
-//	│            Node.MuxRangeFeed            │◄──────────────────────────────────────────────────┐
-//	└─────────────────┬───▲───────────────────┘                  ▲                                │
-//	 Sender.AddStream │   │LockedMuxStream.Send                  │                                │
-//			 ┌────────────▼───┴──────────┐                           │                                │
-//			 │ Buffered/Unbuffered Sender├───────────┐               │                                │
-//			 └────────────┬──────────────┘           │               │                                │
-//			 	   				  │                          │               │                                │
-//			 	   ┌────────▼─────────┐                │               │                                │
-//			 	   │ Stores.Rangefeed │                │               │                                │
-//			 	   └────────┬─────────┘                │               │                                │
-//			 	   				  │                          │               │                                │
-//			 	    ┌───────▼─────────┐         BufferedSender      BufferedSender                      │
-//			 	    │ Store.Rangefeed │ SendUnbuffered/SendBuffered SendBufferedError ─────► BufferedSender.run
-//			 	    └───────┬─────────┘ (catch-up scan)(live raft)     ▲
-//			 	   				  │                        ▲                 │
-//			 	   ┌────────▼──────────┐             │                 │
-//			 	   │ Replica.Rangefeed │             │                 │
-//			 	   └────────┬──────────┘             │                 │
-//			 	   				  │                        │                 │
-//			 	    ┌───────▼──────┐                 │                 │
-//			 	    │ Registration │                 │                 │
-//			 	    └──────┬───────┘                 │                 │
-//			 	    			 │      								   │					  		 │
-//			 	    			 │                         │                 │
-//			 	    			 └─────────────────────────┘─────────────────┘
-//			 	    		BufferedPerRangeEventSink.Send    BufferedPerRangeEventSink.Disconnect
-//
 // BufferedSender is embedded in every rangefeed.BufferedPerRangeEventSink,
 // serving as a helper which buffers events before forwarding events to the
 // underlying gRPC stream.
-//
-// Refer to the comments above UnbufferedSender for more details on the role of
-// senders in the entire rangefeed architecture.
 type BufferedSender struct {
 	// Note that lockedMuxStream wraps the underlying grpc server stream, ensuring
 	// thread safety.
 	sender ServerStreamSender
 
-	// metrics is used to record rangefeed metrics for the node.
-	metrics RangefeedMetricsRecorder
-}
-
-func NewBufferedSender(
-	sender ServerStreamSender, metrics RangefeedMetricsRecorder,
-) *BufferedSender {
-	return &BufferedSender{
-		sender:  sender,
-		metrics: metrics,
+	// queueMu protects the buffer queue.
+	queueMu struct {
+		syncutil.Mutex
+		stopped bool
+		buffer  *eventQueue
 	}
```

Comment:
I will add it in a follow up PR. I didn't it to this PR since I wasn't sure what cap size would be good. I don't want this to be the reason we are not seeing good performance. And we already have memory accounting for these events on the server side.